MatrixSDK/Data/MXRoomSummaryUpdater.m

```diff
@@ -148,4 +161,108 @@ - (BOOL)session:(MXSession *)session updateRoomSummary:(MXRoomSummary *)summary
     return updated;
 }
 
+- (BOOL)session:(MXSession *)session updateRoomSummary:(MXRoomSummary *)summary withServerRoomSummary:(MXRoomSyncSummary *)serverRoomSummary roomState:(MXRoomState *)roomState
+{
+    BOOL updated = NO;
+
+    // Update room members count
+    if (-1 != serverRoomSummary.joinedMemberCount || -1 != serverRoomSummary.invitedMemberCount)
+    {
+        updated |= [self updateSummaryMemberCount:summary session:session withServerRoomSummary:serverRoomSummary roomState:roomState];
+    }
+
+    // Compute display name from summary heroes if there was no name nor canonical alias
+    if (!roomState.name && !roomState.canonicalAlias)
+    {
+        updated |= [self updateSummaryDisplayname:summary session:session withServerRoomSummary:serverRoomSummary roomState:roomState];
+    }
+
+    return updated;
+}
+
+- (BOOL)updateSummaryDisplayname:(MXRoomSummary *)summary session:(MXSession *)session withServerRoomSummary:(MXRoomSyncSummary *)serverRoomSummary roomState:(MXRoomState *)roomState
+{
+    BOOL updated = NO;
+
+    // Compute a non internationalised display name based on
+    // https://docs.google.com/document/d/11i14UI1cUz-OJ0knD5BFu7fmT6Fo327zvMYqfSAR7xs/edit#
+    if (serverRoomSummary.heroes.count == 0 || roomState.membersCount.members <= 1)
+    {
+        summary.displayname = @"Empty Room";
+        updated = YES;
+    }
+    else if (1 <= serverRoomSummary.heroes.count)
+    {
+        NSMutableArray<NSString*> *memberNames = [NSMutableArray arrayWithCapacity:serverRoomSummary.heroes.count];
+        for (NSString *hero in serverRoomSummary.heroes)
+        {
+            NSString *memberName = [roomState.members memberName:hero];
+            if (!memberName)
+            {
+                memberName = hero;
+            }
+
+            [memberNames addObject:memberName];
+        }
+
+        if (memberNames.count == 1)
+        {
+            summary.displayname = memberNames.firstObject;
+        }
+        else
+        {
+            if (serverRoomSummary.heroes.count == summary.membersCount.members - 1)
+            {
+                NSString *lastMemberName = memberNames.lastObject;
+                [memberNames removeLastObject];
+
+                summary.displayname = [NSString stringWithFormat:@"%@ & %@",
+                                       [memberNames componentsJoinedByString:@", "],
+                                       lastMemberName];
+            }
+            else
+            {
+                NSUInteger otherCount = summary.membersCount.members - 1 - serverRoomSummary.heroes.count;
+                summary.displayname = [NSString stringWithFormat:@"%@ & %@ %@",
+                                       [memberNames componentsJoinedByString:@", "],
+                                       @(otherCount),
+                                       (1 < otherCount) ? @"others" : @"other"];
```

Comment:
How to localize "others"/"other" string? Maybe you should add TODO here to track this point

Reply:
Yeap. This could be done in the kit or here with stg like `MXSendReplyEventStringsLocalizable`.